transaction.go

```diff
@@ -369,6 +379,134 @@ func (t *Transaction) Encode() []byte {
 	return mustRLPEncode(&temp)
 }
 
+// DecodeTransactionPayloadMessage returns the signable message for the transaction payload.
+//
+// This message is signed by the authorizers account.
+func DecodeTransactionPayloadMessage(envelopeMessage []byte) (*Transaction, error) {
+	temp := payloadCanonicalForm{}
+	mustRLPDecode(envelopeMessage, &temp)
+	authorizers := make([]Address, len(temp.Authorizers))
+	for i, auth := range temp.Authorizers {
+		authorizers[i] = BytesToAddress(auth)
+	}
+	t := new(Transaction)
+	*t = Transaction{
+		Script:           temp.Script,
+		Arguments:        temp.Arguments,
+		ReferenceBlockID: BytesToID(temp.ReferenceBlockID),
+		GasLimit:         temp.GasLimit,
+		ProposalKey: ProposalKey{
+			Address:        BytesToAddress(temp.ProposalKeyAddress),
+			KeyIndex:       int(temp.ProposalKeyIndex),
+			SequenceNumber: temp.ProposalKeySequenceNumber,
+		},
+		Payer:       BytesToAddress(temp.Payer),
+		Authorizers: authorizers,
+	}
```

Comment:
Maybe just the shorter:

```suggestion
	t := &Transaction{
		Script:           temp.Script,
		Arguments:        temp.Arguments,
		ReferenceBlockID: BytesToID(temp.ReferenceBlockID),
		GasLimit:         temp.GasLimit,
		ProposalKey: ProposalKey{
			Address:        BytesToAddress(temp.ProposalKeyAddress),
			KeyIndex:       int(temp.ProposalKeyIndex),
			SequenceNumber: temp.ProposalKeySequenceNumber,
		},
		Payer:       BytesToAddress(temp.Payer),
		Authorizers: authorizers,
	}
```

Or maybe I'm missing something